app/src/main/java/com/example/jraw_test_2/PaintFragment.java

```diff
@@ -7,37 +7,63 @@
 import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
 
+import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.Button;
 
 public class PaintFragment extends Fragment {
 
     PaintView paintView;
-
-//    public PaintFragment() {
-//
-//    }
-//
-//    public static PaintFragment getInstance() {
-//        PaintFragment fragment = new PaintFragment();
-//        fragment.setRetainInstance(true);
-//        return fragment;
-//    }
+    Button undoButton, clearButton, submitButton;
 
     @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState) {
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         // Inflate the layout for this fragment
         View view = inflater.inflate(R.layout.fragment_paint, container, false);
+        initUI(view);
+        return view;
+    }
 
+    private void initUI(View view) {
         paintView = (PaintView) view.findViewById(R.id.PaintView);
 
-        return view;
+        undoButton = (Button) view.findViewById(R.id.undoButton);
+        undoButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                paintView.undo();
+            }
+        });
+
+        clearButton = (Button) view.findViewById(R.id.clearButton);
+        clearButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                paintView.clear();
+                paintView.invalidate();
+            }
+        });
+
+        submitButton = (Button) view.findViewById(R.id.submitButton);
+        submitButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // TODO: implement paintView.submitDrawing()
+                paintView.submitDrawing();
+                paintView.clear();
+                paintView.invalidate();
+                // TODO: display submission confirmation
```

Comment:
There's a click listener created for each of the three buttons, so they can do different things. Right now, `submitButton` does the same thing as `clearButton` because the method `paintView.submitDrawing()` hasn't been implemented yet.

Reply:
I see, makes sense. I was just testing the pull request review features. Didn't know what to say haha